client/src/Components/PostCreator/PostCreator.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import 'bootstrap/dist/css/bootstrap.css';
+import { Form } from 'react-bootstrap';
+import { useHistory } from 'react-router';
+
+const PostCreator = (props) => {
+	let history = useHistory();
+
+	function handleClick(e) {
+		e.preventDefault();
+		let formElement = document.querySelector('form');
+		var formData = new FormData(formElement);
+		let title = formData.get('titleInput');
+		let body = formData.get('bodyInput');
+		fetch('/posts/add', {
+			headers: {
+				'Content-Type': 'application/json'
+			},
+			method: 'POST',
+			body:JSON.stringify({
+				title,
+				body
+			})
+		});
+		history.push('/');
```

Comment:
What happens if the data fails the validation or the server fails to connect to the DB, for instance. How do you handle these situations?